workspaces/src/network/mod.rs

```diff
@@ -0,0 +1,169 @@
+    /// The helper URL to create top level account out of for certain networks.
+    fn helper_url(&self) -> String;
+}
+
+#[async_trait]
+pub trait TopLevelAccountCreator {
+    async fn create_tla(
+        &self,
+        id: AccountId,
+        signer: InMemorySigner,
+    ) -> anyhow::Result<CallExecution<Account>>;
+
+    async fn create_tla_and_deploy<P: AsRef<Path> + Send + Sync>(
+        &self,
+        id: AccountId,
+        signer: InMemorySigner,
+        wasm: P,
+    ) -> anyhow::Result<CallExecution<Contract>>;
+}
+
+// NOTE: Not all networks/runtimes will have the ability to be able to do dev_deploy.
+// This trait acts as segmented boundary for only specific networks such as sandbox and testnet.
+pub trait AllowDevAccountCreation {}
+
+#[async_trait]
+pub trait DevAccountDeployer {
+    fn dev_generate(&self) -> (AccountId, InMemorySigner);
+    async fn dev_create(&self) -> anyhow::Result<Account>;
+    async fn dev_deploy<P: AsRef<Path> + Send + Sync>(&self, wasm: P) -> anyhow::Result<Contract>;
+}
+
+#[async_trait]
+impl<T> DevAccountDeployer for T
+where
+    T: TopLevelAccountCreator + NetworkInfo + AllowDevAccountCreation + Send + Sync,
+{
+    fn dev_generate(&self) -> (AccountId, InMemorySigner) {
+        let account_id = crate::rpc::tool::random_account_id();
+        let signer =
+            InMemorySigner::from_seed(account_id.clone(), KeyType::ED25519, DEV_ACCOUNT_SEED);
+
+        let mut savepath = self.keystore_path();
+
+        // TODO: potentially make this into the async version:
+        std::fs::create_dir_all(savepath.clone()).unwrap();
+
+        savepath = savepath.join(account_id.to_string());
+        savepath.set_extension("json");
+        signer.write_to_file(&savepath);
+
+        (account_id, signer)
+    }
+
+    async fn dev_create(&self) -> anyhow::Result<Account> {
+        let (account_id, signer) = self.dev_generate();
+        let account = self.create_tla(account_id.clone(), signer).await?;
+        account.into()
+    }
+
+    async fn dev_deploy<P: AsRef<Path> + Send + Sync>(&self, wasm: P) -> anyhow::Result<Contract> {
+        let (account_id, signer) = self.dev_generate();
+        let contract = self
+            .create_tla_and_deploy(account_id.clone(), signer, wasm)
+            .await?;
+        contract.into()
+    }
+}
+
+pub trait AllowStatePatching {}
+
+#[async_trait]
+pub trait StatePatcher {
+    async fn patch_state<U>(
+        &self,
+        contract_id: AccountId,
+        key: String,
+        value: &U,
+    ) -> anyhow::Result<()>
+    where
+        U: BorshSerialize + Send + Sync;
+}
+
+#[async_trait]
+impl<T> StatePatcher for T
+where
+    T: AllowStatePatching + NetworkClient + Send + Sync,
+{
+    async fn patch_state<U>(
+        &self,
+        contract_id: AccountId,
+        key: String,
+        value: &U,
+    ) -> anyhow::Result<()>
+    where
+        U: BorshSerialize + Send + Sync,
+    {
+        let value = U::try_to_vec(value).unwrap();
```

Comment:
Reason for unwrapping (if keeping this pattern)?

Reply:
Didn't keep this pattern, so resolving this one